crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -85,24 +82,18 @@ impl CheckpointStore {
         contents: CheckpointContents,
         epoch_store: &AuthorityPerEpochStore,
     ) {
-        for transaction in contents.iter() {
-            debug!(
-                "Manually inserting genesis transaction in checkpoint DB: {:?}",
-                transaction.transaction
-            );
+        if epoch_store.epoch() == checkpoint.summary.epoch {
```

Comment:
Is genesis checkpoint guaranteed to be from epoch 0?

Reply:
This code does not have this assumption, right? It just check if epoch of genesis checkpoint(whatever it is) is same as epoch of current epoch store

Reply:
Right, I am just curious (and wondering if a stronger assert is needed here)

Reply:
I don't think this particular line need stronger assert. If we think some particular enforcement is needed for genesis, perhaps we can add some kind of `verify` method after loading it that enforces invariants on it.

As far as this code is concerned we just insert some checkpoint, and if it matches current epoch we appropriately update data in checkpoint builder